### PR TITLE
assignment file added

### PR DESCRIPTION
**Pull request by mayurjadhav23**
Problem statement: File Monitoring system

The goal of this task is to design and develop a file monitoring system.
Generate a random writer that writes pseudo strings (with at least 50% probability of generating the “MARUTI” keyword. Note that we are strictly looking at the "MARUTI" keyword only as a single value not the MARUTI keyword inside a random string) into two separate files at regular interval of time.
It should have the capability to monitor both the files and count the total number of occurrences for the “MARUTI” keyword by each file and write output to the "counts.log" file.

### CODE REVIEW

File: FEM.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                    f2.write(random_string(8) + " ")
+            await asyncio.sleep(1)
+
+    async def monitor(self):
+        while True:
+            with open(self.logfile, "a") as log:
+                log.write("File1: " + str(self.count1) + " File2: " + str(self.count2) + " at " + time.ctime() + "\n")
+            await asyncio.sleep(10)
+
+    def start(self):
```
Comment (helidesai8) [inline FEM.py]: @Mayur1Mtech Can we remove this method from the class since its unncessary

File: FEM.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    def __init__(self, file1, file2, logfile):
+        self.file1 = file1
+        self.file2 = file2
+        self.logfile = logfile
+        self.count1 = 0
+        self.count2 = 0
+
+    async def random_writer(self):
+        while True:
+            with open(self.file1, 'a') as f1, open(self.file2, 'a') as f2:
```
Comment (helidesai8) [inline FEM.py]: @Mayur1Mtech Where will these file objects be closed??

File: FEM.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                if random.random() > 0.5:
```
Comment (helidesai8) [inline FEM.py]: @Mayur1Mtech Can we improve this logic? Since it is not right and will not be able to ensure a 50% probability

File: FEM.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                    f1.write("MARUTI ")
+                    self.count1 += 1
+                else:
+                    f1.write(random_string(8) + " ")
```
Comment (helidesai8) [inline FEM.py]: @Mayur1Mtech Why is the string length set to a constant value of 8?

File: FEM.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        self.file2 = file2
+        self.logfile = logfile
+        self.count1 = 0
+        self.count2 = 0
+
+    async def random_writer(self):
+        while True:
+            with open(self.file1, 'a') as f1, open(self.file2, 'a') as f2:
+                if random.random() > 0.5:
+                    f1.write("MARUTI ")
```
Comment (helidesai8) [inline FEM.py]: @Mayur1Mtech Why have we added a space after the string MARUTI

**Review by helidesai8 — CHANGES_REQUESTED** [review]

File: FEM.py (hunk 1 above)
Comment (mayurjadhav23) [reply FEM.py]: it will give error if we remove it from class-"AttributeError: 'FileMonitor' object has no attribute 'start' "

File: FEM.py (hunk 4 above)
Comment (mayurjadhav23) [reply FEM.py]: It not a constant value...we have random_string(length) function to generate pseudo strings..we can generate string of any length

File: FEM.py (hunk 5 above)
Comment (mayurjadhav23) [reply FEM.py]: its just for good visualisation....if we not add space the output will be like MARUTIafgtyhb & WITH Space MARUTI afgtyhb

File: FEM.py (hunk 2 above)
Comment (mayurjadhav23) [reply FEM.py]: ok..I will add it

File: FEM.py (hunk 3 above)
Comment (mayurjadhav23) [reply FEM.py]: I tried that logic:
random.choices(population, weights=50, k=1)
I will try it again

**Review by helidesai8 — APPROVED** [review]

**Pull request merged**